packages/@react-aria/textfield/src/useTextField.ts

```diff
@@ -69,6 +69,7 @@ export function useTextField(
       {
         disabled: isDisabled,
         readOnly: isReadOnly,
+        required: isRequired,
```

Comment:
looks like `aria-required` is discouraged actually
I'm not sure we can/should use both together
@majornista

Reply:
In addition to that, I think another problem is that in the following `form`, the user can submit even though there is no value in input

```jsx
const Form = () => (
  <form>
    <input type="text" aria-required="true" />
    <button type="submit">submit</button>
  </form>
);
```

To avoid this, I think it need `required`.

Reply:
So to make this kind of change, we'd need to stop relying on the css `:invalid` selector
https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/components/textfield/skin.css#L83
And we'd probably need to do this in a lot of places. This might be on spectrum-css' radar, we'd have to ask them. 
We'll also have to run this by Design, but last guidance was that a brand new form should not have any invalid red outlines, that should only appear after the user tries to submit so they have specific guidance on what to fix.
I'll check in with them next week.

Reply:
Got an answer much faster than I thought.
We should not rely on browser validation as it will be inconsistent. Validation is up to the application. That’s why we use aria-required and not required.
If you are using our hooks, you can just pass `required` directly to the element you want it on.
If you are using the Spectrum components, then you should use the validationState prop.